### PR TITLE
Make page load faster by simplifying the user switching logic for Shoppy

### PR DESCRIPTION
**Pull request by heypoom**
We can get rid of the lengthy loading spinner on page load and reload by getting rid of the login, logout and user endpoints and the session middleware, and simply set the "user" cookie on the client side when we want to switch sites.

# Demo

https://www.loom.com/share/fc7eb7f7d9c647d0afdda322005388f3?sid=376cef92-a6ae-466d-b52a-59a972aefc95

### CODE REVIEW

**Comment by heypoom** [comment]
PS. there is a site-breaking bug where the analytics overview page infinitely refreshes. Will work on this.

File: src/components/SiteSwitcher.tsx [hunk 1]
```diff
@@ -14,13 +14,13 @@ import { SiteKey } from "../types/site"
...
-  const loginMutation = useMutation({
-    mutationFn: loginToSite,
+  const switchSiteMutation = useMutation({
+    mutationFn: switchSite,
     mutationKey: ["login"],
   })
 
   async function changeSite(key: SiteKey) {
-    await loginMutation.mutateAsync(key)
+    await switchSiteMutation.mutateAsync(key)
```
Comment (heypoom) [inline src/components/SiteSwitcher.tsx:23]: re-fetches the data (e.g. products & categories) before switching the site.

File: src/index.tsx [hunk 2]
```diff
@@ -26,8 +26,11 @@ import "./styles/viz-workaround.css"
 
 import { store } from "./store"
 
+import { setInitialSiteCookie } from "./utils/switch-site"
 import { ThemeProvider } from "./components/ThemeProvider"
 
+setInitialSiteCookie()
```
Comment (heypoom) [inline src/index.tsx:32]: Set an initial user if the page has never been visited before

File: src/store/site.ts [hunk 3]
```diff
@@ -1,5 +1,15 @@
...
-export const siteAtom = atomWithStorage<SiteKey>("site", "stitch")
+export const SITE_KEY = "site"
+export const DEFAULT_SITE: SiteKey = "stitch"
+
+export const siteAtom = atomWithStorage<SiteKey>(
+  SITE_KEY,
+  DEFAULT_SITE,
+  createJSONStorage(),
+  {
+    getOnInit: true,
```
Comment (heypoom) [inline src/store/site.ts:13]: without `getOnInit: true`, we get a flash of the default theme - this improves performance.

**Review by WiNloSt — APPROVED** [review]
Tested this works great. The performance is really a lot better.

File: api/src/main.ts [hunk 4]
```diff
@@ -18,15 +16,10 @@ setupMiddleware(app)
...
 const router = Router()
 router.get("/", (_, res) => res.send({ status: "ok" }))
-router.post("/login", loginHandler)
-router.post("/logout", logoutHandler)
-router.get("/sso/metabase", metabaseAuthHandler)
+router.get("/sso/metabase", restrict, metabaseAuthHandler)
 router.get("/products", restrict, productListHandler)
 router.get("/categories", restrict, categoryListHandler)
 router.get("/product/:id", productDetailHandler)
-router.get("/user", restrict, (req: any, res: any) => {
```
[the remark this answers is not in the document]
Comment (heypoom) [reply api/src/main.ts:27]: @WiNloSt yep, it hasn't been used for 3 months now 😅

**Pull request merged**